src/notifications/thermalnotifier.cpp

```diff
@@ -14,50 +14,44 @@
 ****************************************************************************/
 #include "notifications/notificationmanager.h"
 #include "notifications/lipsticknotification.h"
-#include "homeapplication.h"
 #include "thermalnotifier.h"
 
-ThermalNotifier::ThermalNotifier(QObject *parent) :
-    QObject(parent),
-    m_thermalState(new DeviceState::Thermal(this)),
-    m_displayState(new DeviceState::DisplayStateMonitor(this)),
-    m_thermalStateNotifiedWhileScreenIsOn(DeviceState::Thermal::Normal)
+#include <dsme/thermalmanager_dbus_if.h>
+
+#include <QDebug>
+
+ThermalNotifier::ThermalNotifier(QObject *parent)
+    : QObject(parent)
+{
+    QDBusConnection::systemBus().connect(QString(),
+                                         thermalmanager_path,
+                                         thermalmanager_interface,
+                                         thermalmanager_state_change_ind,
+                                         this,
+                                         SLOT(applyThermalState(const QString&)));
+}
```

Comment:
I do not see initial value being queried - is that intentional?

Then again I thought the old code did that, but on second glance it probably does not - so no change in logic there.

And I guess this would matter only when booting up frozen or semi-melted device? Or switching user while under such duress.

Reply:
Old code didn't do that and didn't make logic changes.